File: Java/Tests/jdi-uitest-wintests/src/test/java/com/epam/jdi/uitests/testing/unittests/tests/complex/ComboBoxTests.java
package com.epam.jdi.uitests.testing.unittests.tests.complex;

import com.epam.jdi.uitests.core.interfaces.complex.IComboBox;
import com.epam.jdi.uitests.testing.unittests.InitTests;
import com.epam.jdi.uitests.testing.unittests.enums.Metals;
import com.epam.web.matcher.testng.Check;
import org.testng.annotations.BeforeMethod;
import org.testng.annotations.Test;

import java.lang.reflect.Method;
import java.util.List;

import static com.epam.jdi.uitests.core.preconditions.PreconditionsState.isInState;
import static com.epam.jdi.uitests.testing.unittests.enums.Metals.Col;
import static com.epam.jdi.uitests.testing.unittests.enums.Metals.Gold;
import static com.epam.jdi.uitests.testing.unittests.enums.Preconditions.METALS_AND_COLORS_PAGE;
import static com.epam.jdi.uitests.testing.unittests.pageobjects.EpamJDIApplication.metalsColorsPage;
import static com.epam.jdi.uitests.testing.unittests.tests.complex.CommonActionsData.*;
import static com.epam.web.matcher.junit.Assert.exception;
import static com.epam.web.matcher.testng.Assert.areEquals;
import static com.epam.web.matcher.testng.Assert.listEquals;
import static java.util.Arrays.asList;

/**
 * Created by Roman_Iovlev on 9/15/2015.
 */
public class ComboBoxTests extends InitTests {
    private static final List<String> oddOptions = asList("Col", "Gold", "Silver", "Bronze", "Selen");

    private IComboBox<Metals> metals() {
        return metalsColorsPage.comboBox;
    }

    @BeforeMethod
    public void before(Method method) {
        isInState(METALS_AND_COLORS_PAGE, method);
    }

    @Test
    public void selectStringTest() {
        metals().select("Gold");
        checkAction("Metals: value changed to Gold");
    }

    @Test
    public void selectIndexTest() {
        metals().select(3);
        checkAction("Metals: value changed to Silver");
    }

    @Test
    public void selectEnumTest() {
        metals().select(Gold);
        checkAction("Metals: value changed to Gold");
    }

    @Test
    public void getOptionsTest() {
        listEquals(metals().getOptions(), oddOptions);
    }

    @Test
    public void getNamesTest() {
        listEquals(metals().getNames(), oddOptions);
    }

    @Test
    public void getValuesTest() {
        listEquals(metals().getValues(), oddOptions);
    }

    @Test
    public void getOptionsAsTextTest() {
        areEquals(metals().getOptionsAsText(), "Col, Gold, Silver, Bronze, Selen");
    }

    @Test
    public void setValueTest() {
        metals().setValue("Blue");
        looseFocus();
        checkAction("Metals: value changed to Blue");
    }

    // Fails
    @Test
    public void getSelectedTest() {
        metals().select("Gold");
        areEquals(metals().getSelected(), "Gold");
    }

    @Test
    public void getSelectedIndexTest() {
        checkActionThrowError(() -> metals().getSelectedIndex(), noElementsMessage); // isDisplayed not defined
    }

    @Test
    public void isSelectedTest() {
        areEquals(metals().isSelected("Col"), true);
    }

    @Test
    public void isSelectedEnumTest() {
        areEquals(metals().isSelected(Col), true);
    }

    @Test
    public void waitSelectedTest() {
        try {
            metals().waitSelected("Col");
        } catch (Exception ex) {
            throw exception("WaitSelected throws exception");
        }
    }

    @Test
    public void waitSelectedEnumTest() {
        new Check("WaitSelected")
            .hasNoExceptions(() -> metals().waitSelected(Col));
    }

    @Test
    public void getValueTest() {
        areEquals(metals().getValue(), "Col");
    }
}

File: Python/Test/jdi_uitests_webtests/test/complex/combo_box_tests.py
from JDI.jdi_assert.testing.assertion import Assert
from Test.jdi_uitests_webtests.main.enums.entities import Metals
from Test.jdi_uitests_webtests.main.enums.preconditions import Preconditions
from Test.jdi_uitests_webtests.main.page_objects.epam_jdi_site import EpamJDISite
from Test.jdi_uitests_webtests.main.utils.common_action_data import CommonActionsData
from Test.jdi_uitests_webtests.test.init_tests import InitTests


class ComboBoxTests(InitTests):

    add_options = ["Col", "Gold", "Silver", "Bronze", "Selen"]
    add_options_string = "Col, Gold, Silver, Bronze, Selen"

    combo_box = EpamJDISite.metals_colors_page.combo_box

    def setUp(self):
        Preconditions.METALS_AND_COLORS_PAGE.is_in_state()

    def test_select_string(self):
        self.combo_box.select("Gold")
        CommonActionsData.check_action("Metals: value changed to Gold")

    def test_select_index(self):
        self.combo_box.select(3)
        CommonActionsData.check_action("Metals: value changed to Silver")

    def test_select_enum(self):
        self.combo_box.select(Metals.GOLD)
        CommonActionsData.check_action("Metals: value changed to Gold")

    def test_get_options(self):
        Assert.assert_equal(self.combo_box.get_options(), self.add_options)

    def test_get_names(self):
        Assert.assert_equal(self.combo_box.get_names(), self.add_options)

    def test_get_values(self):
        Assert.assert_equal(self.combo_box.get_values(), self.add_options)

    def test_get_options_as_text(self):
        Assert.assert_equal(self.combo_box.get_options_as_text(), self.add_options_string)

    def test_set_value_text(self):
        self.combo_box.set_value("Blue")
        CommonActionsData.loose_focus()
        CommonActionsData.check_action("Metals: value changed to Blue")

    def test_get_selected(self):
        self.combo_box.select("Gold")
        Assert.assert_equal(self.combo_box.get_selected(), "Gold")

    def test_get_selected_index(self):
        CommonActionsData.check_action_throw_error(lambda: self.combo_box.get_selected_index(), CommonActionsData.no_elements_message())

    def test_is_selected(self):
        Assert.assert_equal(self.combo_box.is_selected("Col"), True)

    def test_is_selected_enum(self):
        Assert.assert_equal(self.combo_box.is_selected(Metals.COL), True)

    def test_get_value(self):
        Assert.assert_equal(self.combo_box.get_value(), "Col")